.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,52 @@
+name: CodeQL Analysis
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 8 * * *'
+
+jobs:
+  analyze:
+    name: CodeQL Analysis
+    runs-on: windows-latest
+    steps:
+      - name: Checkout repository
+        id: checkout_repo
+        uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
```

Comment:
missing setup dotnet step https://github.com/microsoft/kiota/blob/3054610dd7e679a099eb2b7fdd9bc6227944cfaa/.github/workflows/abstractions-dotnet.yml#L17

Reply:
Shouldn't be required, because the GH Ubuntu runner comes with [.NET SDKs preinstalled](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#net-core-sdk).
We just tested the pipeline on `ubuntu-latest` with `dotnet build` and it works flawlessly 🙂

Reply:
I usually always add the dotnet install step because relying on pre-installed tools is a moving target (e.g. what are going to be the defaults when dotnet 6 comes out? will that derail this worklow?). Worst case it's a no-op (besides setting environment variables), best case it prevents workflows from stopping to work when new versions come out or default images change. But I get it's a preference thing, so I won't block on that.

Reply:
That makes perfect sense, actually. 
We'd add it promptly.